src/core/utils/stringutils.cpp

```diff
@@ -161,3 +162,13 @@ const QStringList StringUtils::csvToStringList( const QString &string )
   values << value;
   return std::move( values );
 }
+
+QString StringUtils::replaceFilenameTags( const QString &string, const QString &filename )
+{
+  // IMPORTANT: an equivalent Java function QFieldUtils on Android needs to be kept synchronized
+  QString replacedString = string;
+  QFileInfo fi( filename );
+  replacedString.replace( QStringLiteral( "#~filename~#" ), fi.fileName() );
+  replacedString.replace( QStringLiteral( "#~extension~#" ), fi.completeSuffix() );
```

Comment:
why don't we use either moustaches `{filename}` or dollar preix-suffix `$filename$` as common and well known template placeholders?

Reply:
@suricactus , no strong feelings, makes sense to me.

@m-kuhn , thoughts?

Reply:
Yes!

Reply:
OK, let's settle for "{filename}" then. The $variant$ might be confusing as QGIS' expression engine do make use of the dollar sign with e.g. $id, $geometry, etc.